### PR TITLE
improve performance and memory usage

### PR DESCRIPTION
**Pull request by MaggieQi**

### CODE REVIEW

File: AnnService/src/Core/SPANN/SPANNIndex.cpp [hunk 1]
```diff
@@ -203,7 +203,7 @@ namespace SPTAG
 
             m_index->SearchIndex(*p_queryResults);
             
-            if (m_pQuantizer.get() != m_index->m_pQuantizer.get()) { p_queryResults->SetTarget(p_queryResults->GetTarget(), m_index->m_pQuantizer); }
+            //if (m_pQuantizer.get() != m_index->m_pQuantizer.get()) { p_queryResults->SetTarget(p_queryResults->GetTarget(), m_index->m_pQuantizer); }
```
Comment (PhilipBAdams) [inline AnnService/src/Core/SPANN/SPANNIndex.cpp:206]: Let's just delete, and if we need to support different head and SSD quantizer, do it in a different PR

File: AnnService/src/Helper/VectorSetReaders/TxtReader.cpp [hunk 2]
```diff
@@ -66,12 +66,12 @@ TxtVectorReader::LoadFile(const std::string& p_filePaths)
             return ErrorCode::FailedOpenFile;
         }
 
-        std::uint32_t fileTaskCount = 1;
+        std::uint32_t fileTaskCount = 0;
```
Comment (PhilipBAdams) [inline AnnService/src/Helper/VectorSetReaders/TxtReader.cpp:69]: I don't understand this change - isn't it reset either way?

**Review by PhilipBAdams — APPROVED** [review]

File: AnnService/inc/SSDServing/SSDIndex.h [hunk 3]
```diff
@@ -134,7 +134,6 @@ namespace SPTAG {
                             double endTime = threadws.getElapsedMs();
                             p_index->DebugSearchDiskIndex(p_results[index], p_internalResultNum, p_internalResultNum, &(p_stats[index]));
                             double exEndTime = threadws.getElapsedMs();
-                            p_results[index].ClearTmp();
```
[the remark this answers is not in the document]
Comment (MaggieQi) [reply AnnService/inc/SSDServing/SSDIndex.h:137]: In the SetTarget, you have already allocated the memory for all quantized targets. So we deallocate the memory does not reduce the total memory usage?

File: AnnService/src/Core/SPANN/SPANNIndex.cpp (hunk 1 above)
Comment (MaggieQi) [reply AnnService/src/Core/SPANN/SPANNIndex.cpp:206]: sure

File: AnnService/src/Helper/VectorSetReaders/TxtReader.cpp (hunk 2 above)
Comment (MaggieQi) [reply AnnService/src/Helper/VectorSetReaders/TxtReader.cpp:69]: yes, so I just recover the previous code here.

File: AnnService/inc/Helper/StringConvert.h [hunk 4]
```diff
@@ -95,7 +95,7 @@ inline bool ConvertStringToUnsignedInt(const char* p_str, DataType& p_value)
         return false;
     }
 
-    if (val < (std::numeric_limits<DataType>::min)() || val >(std::numeric_limits<DataType>::max)())
```
[the remark this answers is not in the document]
Comment (MaggieQi) [reply AnnService/inc/Helper/StringConvert.h:98]: yes, so it will check whether the uint64_t we get will overflow or not. When it will be negative? This function will only be called when the DataType is uint8_t, uint16_t, uint32_t and uint64_t in the function ConvertStringTo<XXX>.

[the remark this answers is not in the document]
Comment (PhilipBAdams) [reply AnnService/inc/Helper/StringConvert.h:98]: I mean if the string is negative

Sample of the behavior I think doesn't make sense
Code:
```C++
template <typename DataType>
void TestUnsignedConv(const std::string & s)
{
  DataType v = 0;
  int a = 0;
  bool success = ConvertStringToUnsignedInt<DataType>(s.c_str(), v);
  std::cout << "Converted  \""<< s << "\" to value " << static_cast<unsigned int>(v) << ". Success: " << success << std::endl;
}

int main() {
  TestUnsignedConv<std::uint8_t>("128");
  TestUnsignedConv<std::uint8_t>("257");
  TestUnsignedConv<std::uint8_t>("-1");
  TestUnsignedConv<std::uint8_t>("-18446744073709551600");
}
```
Output:
```
Converted  "128" to value 128. Success: 1
Converted  "257" to value 0. Success: 0
Converted  "-1" to value 0. Success: 0
Converted  "-18446744073709551600" to value 16. Success: 1
```
It is because `std::strtoull` will do the overflow, if it overflows into range the result is unintuitive, since it overflows based on the limits of `std::uint64_t` instead of `DataType`.

File: AnnService/inc/SSDServing/SSDIndex.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (PhilipBAdams) [reply AnnService/inc/SSDServing/SSDIndex.h:137]: When vectorset is provided, after the search is complete we will load the full-dim vectors into memory. With ClearTmp feature, quantized query memory will have been freed. Without it, we need the memory for both ADC tables and the reranking vectors. 

Alternately, we can do the free manually after in a loop after `SearchSequential`, instead of inside `SearchSequential` logic

**Review by PhilipBAdams — APPROVED** [review]
